lib/buckets/common.js

```diff
@@ -53,8 +62,24 @@ var resemblesIpRegexStr = format('^%s\.%s\.%s\.%s$', threeDigitRegexStr,
 var resemblesIpRegex = new RegExp(resemblesIpRegexStr);
 
 function isValidBucketName(name) {
-    return bucketRegex.test(name) && !resemblesIpRegex.test(name) &&
-        name.length >= 3 && name.length <= 63;
+    assert.string(name, 'name');
+
+    return (bucketRegex.test(name) && !resemblesIpRegex.test(name) &&
+        !nullRegex.test(name) &&
+        Buffer.byteLength(name) >= MIN_BUCKET_NAME_BYTE_SIZE &&
+        Buffer.byteLength(name) <= MAX_BUCKET_NAME_BYTE_SIZE);
```

Comment:
Actually AFAIK the *bucket name* length restriction *is* about characters and not bytes. That's per https://tools.ietf.org/html/rfc1035#section-2.3.1

```
The labels must follow the rules for ARPANET host names.  They must
start with a letter, end with a letter or digit, and have as interior
characters only letters, digits, and hyphen.  There are also some
restrictions on the length.  Labels must be 63 characters or less.
```

Reply:
Interesting... will make the change.